### ISSUE TITLE
PistonMovement: CLAIMS_ONLY no longer allows sticky piston use

### ISSUE THREAD

**Issue by mrcoffee1026** [GriefPrevention/GriefPrevention#1783]
### Observed Behavior

Pistons and ONLY regular pistons function within claims.  Sticky pistons function no where on the server.

### Expected Behavior

Sticky pistons should also function inside claims.

### Reproduction steps

Attempt sticky piston use inside a claim with
PistonMovement: CLAIMS_ONLY
option set.

### Stack trace or error log

```shell
No errors.
```

### Server version

```shell
This server is running Paper version git-Paper-176 (MC: 1.18.1) (Implementing API version 1.18.1-R0.1-SNAPSHOT) (Git: b164899)
You are 5 version(s) behind
Download the new version at: https://papermc.io/downloads
Previous version: git-Paper-168 (MC: 1.18.1)
```

### GriefPrevention version

```shell
GriefPrevention version 16.17.1-42-g3562a23-dirty
```

### Configuration

```yaml
# Default values are perfect for most servers.  If you want to customize and have a question, look for the answer here first: http://dev.bukkit.org/bukkit-plugins/grief-prevention/pages/setup-and-configuration/

GriefPrevention:
  SeaLevelOverrides:
    world: -1
    world_nether: -1
    world_the_end: -1
    world_tardis_siluria: -1
    world_tardis_skaro: -1
    world_tardis_gallifrey: -1
    tardis_zero_room: -1
    nether_claims: -1
    pvp: -1
    tardis_timevortex: -1
    QuirpyEnd: -1
    creative: -1
    pixelmap: -1
    pixelmaptwo: -1
    staff_creative: -1
    event: -1
    EnderFarm: -1
    RolePlay: -1
  Claims:
    Mode:
      RolePlay: Survival
      pixelmaptwo: Disabled
      pvp: Disabled
      EnderFarm: Disabled
      world_nether: Disabled
      world_tardis_skaro: Disabled
      world_tardis_siluria: Disabled
      creative: Disabled
      event: Survival
      tardis_timevortex: Disabled
      pixelmap: Survival
      world: Survival
      QuirpyEnd: Survival
      world_tardis_gallifrey: Disabled
      staff_creative: Disabled
      tardis_zero_room: Disabled
      world_the_end: Disabled
      nether_claims: Survival
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: false
    LockTrapDoors: false
    LockFenceGates: false
    EnderPearlsRequireAccessTrust: true
    RaidTriggersRequireBuildTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 500
    Claim Blocks Accrued Per Hour:
      Default: 20
    Max Accrued Claim Blocks:
      Default: 100000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 4
    AutomaticNewPlayerClaimsRadiusMinimum: 0
    ExtendIntoGroundDistance: 383
    MinimumWidth: 5
    MinimumArea: 100
    MaximumDepth: 0
    InvestigationTool: WOODEN_SWORD
    ModificationTool: GOLDEN_SHOVEL
    Expiration:
      ChestClaimDays: 7
      UnusedClaimDays: 14
      AllClaims:
        DaysInactive: 90
        ExceptWhenOwnerHasTotalClaimBlocks: 5000
        ExceptWhenOwnerHasBonusClaimBlocks: 2500
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 0
    CreationRequiresWorldGuardBuildPermission: true
    VillagerTradingRequiresPermission: true
    CommandsRequiringAccessTrust: /sethome
    DeliverManuals: false
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true
  Spam:
    Enabled: false
    LoginCooldownSeconds: 60
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/global;/local
    WhisperSlashCommands: /tell;/pm;/r;/whisper;/msg
    WarningMessage: Please reduce your noise level.  Spammers will be banned.
    BanOffenders: false
    BanMessage: Banned for spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 120
    Logout Message Delay In Seconds: 0
  PvP:
    RulesEnabledInWorld:
      world: true
      world_nether: true
      world_the_end: false
      world_tardis_siluria: true
      world_tardis_skaro: true
      world_tardis_gallifrey: true
      tardis_zero_room: false
      nether_claims: true
      pvp: true
      tardis_timevortex: true
      QuirpyEnd: true
      creative: false
      pixelmap: false
      pixelmaptwo: false
      staff_creative: false
      event: false
      EnderFarm: false
      RolePlay: true
    ProtectFreshSpawns: true
    PunishLogout: true
    CombatTimeoutSeconds: 15
    AllowCombatItemDrop: false
    BlockedSlashCommands: /home;/vanish;/spawn;/tpa;/fly;/speed;/mc
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: false
      AdministrativeClaims: false
      AdministrativeSubdivisions: false
    AllowLavaDumpingNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    AllowFlintAndSteelNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    ProtectPetsOutsideLandClaims: false
  Economy:
    ClaimBlocksMaxBonus: 0
    ClaimBlocksPurchaseCost: 3.0
    ClaimBlocksSellValue: 3.0
  ProtectItemsDroppedOnDeath:
    PvPWorlds: false
    NonPvPWorlds: false
  BlockLandClaimExplosions: false
  BlockSurfaceCreeperExplosions: false
  BlockSurfaceOtherExplosions: false
  LimitSkyTrees: false
  LimitTreeGrowth: false
  PistonMovement: CLAIMS_ONLY
  PistonExplosionSound: true
  FireSpreads: true
  FireDestroys: false
  AdminsGetWhispers: true
  AdminsGetSignNotifications: false
  SmartBan: false
  Mute New Players Using Banned Words: false
  MaxPlayersPerIpAddress: 3
  SilenceBans: true
  Siege:
    Worlds: []
    BreakableBlocks:
    - GRASS_BLOCK
    - DIRT
    - COBBLESTONE
    - OAK_PLANKS
    - SPRUCE_PLANKS
    - BIRCH_PLANKS
    - JUNGLE_PLANKS
    - ACACIA_PLANKS
    - DARK_OAK_PLANKS
    - SAND
    - GRAVEL
    - GLASS
    - GRASS
    - FERN
    - DEAD_BUSH
    - WHITE_WOOL
    - ORANGE_WOOL
    - MAGENTA_WOOL
    - LIGHT_BLUE_WOOL
    - YELLOW_WOOL
    - LIME_WOOL
    - PINK_WOOL
    - GRAY_WOOL
    - LIGHT_GRAY_WOOL
    - CYAN_WOOL
    - PURPLE_WOOL
    - BLUE_WOOL
    - BROWN_WOOL
    - GREEN_WOOL
    - RED_WOOL
    - BLACK_WOOL
    - SNOW
    - GLASS_PANE
    DoorsOpenDelayInSeconds: 300
    CooldownEndInMinutes: 60
  EndermenMoveBlocks: false
  SilverfishBreakBlocks: false
  CreaturesTrampleCrops: false
  RabbitsEatCrops: false
  HardModeZombiesBreakDoors: false
  Database:
    URL: jdbc:mysql://localhost:3306/griefprevention?useSSL=false
    UserName: ****
    Password: ****
  UseBanCommand: false
  BanCommandPattern: ban %name% %reason%
  Advanced:
    fixNegativeClaimblockAmounts: true
    ClaimExpirationCheckRate: 60
    OfflinePlayer_cache_days: 90
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: false
      Suspicious Activity: false
      Administrative Activity: false
      Debug: false
      Muted Chat Messages: false
```

### Running without GriefPrevention

- [ ] I attempted running the server without GriefPrevention installed.
- [ ] The problem does _not_ occur when GriefPrevention is removed from the server.

### Running with only GriefPrevention

- [ ] I attempted running only GriefPrevention on the server.
- [ ] The issue still occurs when GriefPrevention is the only plugin running.

### Running on a fresh, clean server installation

- [ ] I attempted testing for the issue on a new server.
- [ ] The issue still occurs on a new server.

### Using unmodified client

- [ ] I attempted testing for the issue with the vanilla client.
- [ ] The issue still occurs when using the vanilla client.

### We appreciate you taking the time to fill out a bug report!

- [X] I searched for similar issues before submitting this bug report.

**Comment by Jikoo** [comment]
Please reproduce using an actual GP build. I'm not sure where you're downloading that, but there is no commit g3562a23 in GP's history, and there are further modifications made to it on top of whatever that is. Unable to reproduce on 16.18-RC1.

**Comment by mrcoffee1026** [comment]
This is the build that comes from this page:
https://github.com/TechFortress/GriefPrevention/releases
first set of assets.
Release calls itself
GriefPrevention version 16.17.1-42-g3562a23-dirty
When doing /versions griefprevention.  But whatever man i'm not going to argue with you about what you guys have chosen to report internally as your release names just so you can close tickets.  After several minutes of changing nothing... the problem went away on its own.  The original claim I was testing in erased itself somehow, I couldn't ever get the one piston on, one piston off effect again even in a new claim... and the original player to report having the problem said it went away for them as well.  All this without restarting the server or changing a single setting.

**Comment by mrcoffee1026** [comment]
AHHA: verified.  It's the y-value.  Pistons work at positive y-values, fail in the negative.  I believed that
ExtendIntoGroundDistance: 383
would be all-encompassing but it apparently is not?  Tried various values for "MaximumDepth:'" but nothing had any effect so I'm assuming its not related in any way, gp just ignores negative y-values it seems.

**Comment by Jikoo** [comment]
> This is the build that comes from this page:
TechFortress/GriefPrevention/releases
first set of assets.
Release calls itself
GriefPrevention version 16.17.1-42-g3562a23-dirty

It's not though. The 16.17.1 release is `16.17.1-2b69191` - I just downloaded it myself to verify.

> AHHA: verified. It's the y-value. Pistons work at positive y-values, fail in the negative. I believed that
> ExtendIntoGroundDistance: 383
> would be all-encompassing but it apparently is not? Tried various values for "MaximumDepth:'" but nothing had any effect so I'm assuming its not related in any way, gp just ignores negative y-values it seems.

Update to either 16.18-RC1 or (preferably, as it fixes another know issue) the most recent dev build off the CI, then edit maximum depth in your config to at or below the min world height.

**Comment by mrcoffee1026** [comment]
The latest build is here:
![latest_release1](https://user-images.githubusercontent.com/3322491/152093283-ab952d7e-466f-4a29-bc3b-d23969e101db.jpg)
![latest_release2](https://user-images.githubusercontent.com/3322491/152093289-9c018875-e5b2-4820-bde7-c29c3b7778d2.jpg)
It calls itself GriefPrevention version 16.17.1-42-g3562a23-dirty
2/3 links send me to this from github.  Anyway I tried the release direct from bukkit.  Pistons work at negative y values... in the sense that they are always allowed whether they are in a claim or not.  There is a notice that states they shouldn't be able to move blocks... but they aren't supposed to be able to be activated at ALL outside a claim, the release fails to prevent this as normal in the release version.

**Comment by mrcoffee1026** [comment]
GriefPrevention version 16.17.1-2b69191

Sticky piston pulling at negative y values also do not work.

**Comment by Jikoo** [comment]
> The latest build is here: ![latest_release1](https://user-images.githubusercontent.com/3322491/152093283-ab952d7e-466f-4a29-bc3b-d23969e101db.jpg) ![latest_release2](https://user-images.githubusercontent.com/3322491/152093289-9c018875-e5b2-4820-bde7-c29c3b7778d2.jpg) It calls itself GriefPrevention version 16.17.1-42-g3562a23-dirty 2/3 links send me to this from github. Anyway I tried the release direct from bukkit. Pistons work at negative y values... in the sense that they are always allowed whether they are in a claim or not. There is a notice that states they shouldn't be able to move blocks... but they aren't supposed to be able to be activated at ALL outside a claim, the release fails to prevent this as normal in the release version.

Interesting, I wonder what the heck Robo uploaded then. I apologize, the commit hash is part of the version, so problems on a commit that doesn't exist generally means not our problem.

Now, your actual issue:
I'm a bit unclear on what exactly pistons are capable of, getting mixed impressions. Generally GP allows pistons that do not move blocks other than the piston itself to function for performance reasons I believe, so if the pistons are functional while not moving blocks that is intended behavior.

You need a dev build (either the 16.18-RC1 or, ideally due to the fix of automatic claim extension during building, the latest build off of the CI) to support negative Y values. You also need to edit `MaximumDepth` to a value at or below the depth of your lowest world, which is likely -64 unless you're running a datapack that changes that.

**Comment by Jikoo** [comment]
For reference, the CI build is [here](https://ci.appveyor.com/project/RoboMWM39862/griefprevention/builds/42088141/artifacts), though it's due to be deleted in 3 days due to AppVeyor's build expiration policy, which means I get to ~~suffer~~ redirect people to abusing JitPack as a CI soon until Robo resurfaces.

**Comment by mrcoffee1026** [comment]
Alright I set the maximumdepth to -64 but again, did not notice any change in behavior by having done that.  Currently, if we stand at a negative y-value within a claim we are able to highlight the claim with the appropriate tool, BUT... when using commands for the claim (such as /trustlist) it says to "stand inside the claim you are curious about") as though we are not in a claim.  I can activate pistons WITHOUT blocks at that level but cannot move blocks even though I am inside of a claim at that level -- note screenshot:
![Screenshot_1](https://user-images.githubusercontent.com/3322491/152209525-fab69bda-4450-4023-8e26-df4994c09671.jpg)


The Secondary issue I mentioned above is that this option: "PistonMovement: CLAIMS_ONLY" is supposed to disable ALL piston movement outside of claims.  It does this... but only on a y+ value.  On a y- value, piston movement is allowed everywhere... even outside of a claim.  However... as you noted: "Generally GP allows pistons that do not move blocks other than the piston itself"  appears to be true... even when they should be able to move blocks inside of a claim in a y- value.

Currently, pistons work properly in a y+ value.  Pistons do not function at all outside of claims, moving blocks or not.
 
* I am using the actual release now, thanks for the clarification!

**Comment by Jikoo** [comment]
The dev build I linked fixed a bug in 16.18-RC1 with claims not auto-extending downwards to include newly placed blocks. Please try again on that dev build with freshly-placed pistons.

Your existing claims that are above y-64 claim depth will not extend down on RC1, and 16.17.1 doesn't really support negative Y at all under any circumstances.

**Comment by mrcoffee1026** [comment]
My apologoies, I thought you were just giving me another link to track more recent builds, not that you were saying this one remedied this issue.  I verified that GriefPrevention version 16.18-RC1-2-g0e3a2e9 does resolve this, thank you much!
